File: aoc2019/day1.hs
getNumbers :: String -> [Int]
getNumbers str = map (read :: String -> Int) $ words str

-- round(num / 3) - 2
calcFuel :: Int -> Int
calcFuel x =
  if (div x 3 - 2) > 0
    then div x 3 - 2
    else 0

-- part 2
recursiveFuel :: Int -> Int
recursiveFuel 0 = 0
recursiveFuel x = calcFuel x + recursiveFuel (calcFuel x)

main :: IO ()
main = do
  content <- readFile "aoc2019/inputs/1.txt"
  let numberInput = getNumbers content
  print $ sum [calcFuel x | x <- numberInput]
    --part 2
  print $ sum [recursiveFuel x | x <- numberInput]

File: aoc2019/day1.py
nums = [int(num.strip()) for num in open('aoc2019/inputs/1.txt').readlines()]


def calc_fuel(mass):
    return (mass//3)-2


def rec_fuel(mass):
    fuel = calc_fuel(mass)
    if fuel < 0:
        return 0
    else:
        return fuel + rec_fuel(fuel)


total_fuel = sum([calc_fuel(mass) for mass in nums])
print(total_fuel)

total_rec_fuel = sum([rec_fuel(mass) for mass in nums])
print(total_rec_fuel)
